lib/killbill_client/models/subscription.rb

```diff
@@ -132,6 +132,55 @@ def update_bcd(user = nil, reason = nil, comment = nil, effective_from_date = ni
       end
 
 
+      #
+      # Block a Subscription
+      #
+      def block_subscription(subscription_id, state_name, service, block_change, block_entitlement, block_billing, requested_date = nil, user = nil, reason = nil, comment = nil, options = {})
+
+        body = KillBillClient::Model::BlockingStateAttributes.new
+        body.state_name = state_name
+        body.service = service
+        body.block_change = block_change
+        body.block_entitlement = block_entitlement
+        body.block_billing = block_billing
+        body.type = "SUBSCRIPTION"
+
+        params = {}
+        params[:requestedDate] = requested_date unless requested_date.nil?
+
+        self.class.put "#{KILLBILL_API_ENTITLEMENT_PREFIX}/#{subscription_id}/block",
+                       body.to_json,
+                       params,
+                       {
+                           :user    => user,
+                           :reason  => reason,
+                           :comment => comment,
+                       }.merge(options)
+      end
+
+      #
+      # Create an entitlement with addOn products
+      #
+      def create_entitlement_with_addOn(entitlements,requested_date, entitlement_date, billing_date, migrated = false, call_completion = false, callTimeoutSec = 3, user = nil, reason = nil, comment = nil, options = {})
```

Comment:
This is already implemented in `lib//killbill_client/models/bulk_subscription.rb`. Easiest to check for these is to `grep` the path suffix (e.g. `createEntitlementWithAddOns` in this case).

Reply:
There two different API calls named very similar for this:  createEntitlementsWithAddOns (Entitlements with 's') and  createEntitlementWithAddOns (Entitlement without 's'). The second one was missed.

Reply:
Oops, indeed! 👍